renderer/components/DNav.tsx

```diff
@@ -131,25 +160,10 @@ class DNavComponent extends React.Component {
   }
 }
 
-DNavComponent.propTypes = {
-  connectionStatus: PropTypes.bool.isRequired,
-  runtimeStatus: PropTypes.bool.isRequired,
-  masterStatus: PropTypes.bool.isRequired,
-  isRunningCode: PropTypes.bool.isRequired,
-  ipAddress: PropTypes.string.isRequired,
-  startTour: PropTypes.func.isRequired,
-  runtimeVersion: PropTypes.string.isRequired,
-  robotState: PropTypes.number.isRequired,
-  heart: PropTypes.bool.isRequired,
-  fieldControlStatus: PropTypes.bool.isRequired,
-  blueMaster: PropTypes.number.isRequired,
-  goldMaster: PropTypes.number.isRequired,
-};
-
-const mapStateToProps = state => ({
+const mapStateToProps = (state: ApplicationState) => ({
   masterStatus: state.fieldStore.masterStatus,
-  blueMaster: state.fieldStore.blueMaster,
-  goldMaster: state.fieldStore.goldMaster,
+  blueMasterTeamNumber: state.fieldStore.blueMasterTeamNumber,
+  goldMasterTeamNumber: state.fieldStore.goldMasterTeamNumber,
   robotState: state.info.robotState,
   heart: state.fieldStore.heart,
   ipAddress: state.info.ipAddress,
```

Comment:
1. info has a type of never. (I thought I fixed this last meeting but I guess it hasn't...?)
2. blueMasterTeamNumber and goldMasterTeamNumber is redlining because it is not in the given state type `FieldState` in `fieldStore.ts`. It is included in actions, but I think `combineReducers` only combines the states, so I am confused on how to fix this.

Reply:
Hmm, ok let me take a look at this.

Reply:
@HaileyJang Ok, some interesting things:
1. So I'm not sure why some of the other reducers were not `never`, but if you type the return type of the reducer, then `combineReducers` will for sure know the type. I guess `combineReducers` sometimes has a hard figuring out what the return types of the reducers should be. (https://github.com/pioneers/dawn/pull/31/commits/1f495abecdfc683bf7e733beefffbd8ac063f537)
2. I think since we access these 2 fields like `state.fieldStore.*`, we can just add to the `FieldState`. (https://github.com/pioneers/dawn/pull/31/commits/375acfe6b29e0a565a487cfc95fc1e4c369f0a69)